File: mycollab-services/src/main/java/com/esofthead/mycollab/module/file/domain/criteria/FileSearchCriteria.java
/**
 * This file is part of mycollab-services.
 *
 * mycollab-services is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-services is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-services.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.module.file.domain.criteria;

import com.esofthead.mycollab.core.arguments.SearchCriteria;
import com.esofthead.mycollab.module.ecm.domain.ExternalDrive;

/**
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 *
 */
public class FileSearchCriteria extends SearchCriteria {
	private static final long serialVersionUID = 1L;

	private String rootFolder;
	private String fileName;
	private String baseFolder;
	private String storageName;
	private ExternalDrive externalDrive;

	public String getFileName() {
		return fileName;
	}

	public void setFileName(String fileName) {
		this.fileName = fileName;
	}

	public String getBaseFolder() {
		return baseFolder;
	}

	public void setBaseFolder(String baseFolder) {
		this.baseFolder = baseFolder;
	}

	public String getRootFolder() {
		return rootFolder;
	}

	public void setRootFolder(String rootFolder) {
		this.rootFolder = rootFolder;
	}

	public String getStorageName() {
		return storageName;
	}

	public void setStorageName(String storageName) {
		this.storageName = storageName;
	}

	public ExternalDrive getExternalDrive() {
		return externalDrive;
	}

	public void setExternalDrive(ExternalDrive externalDrive) {
		this.externalDrive = externalDrive;
	}
}

File: mycollab-services/src/main/java/com/mycollab/module/file/domain/criteria/FileSearchCriteria.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.module.file.domain.criteria

import com.mycollab.db.arguments.SearchCriteria
import com.mycollab.module.ecm.domain.ExternalDrive

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
class FileSearchCriteria : SearchCriteria() {

    var rootFolder: String? = null
    var fileName: String? = null
    var baseFolder: String? = null
    var storageName: String? = null
    var externalDrive: ExternalDrive? = null
}
